selfservice/strategy/passkey/passkey_schema_extension.go

```diff
@@ -0,0 +1,50 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package passkey
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/ory/jsonschema/v3"
+	"github.com/ory/kratos/identity"
+	"github.com/ory/kratos/schema"
+)
+
+type SchemaExtension struct {
+	Identifier string
+	sync.Mutex
+}
+
+func (e *SchemaExtension) Run(_ jsonschema.ValidationContext, s schema.ExtensionConfig, value any) error {
+	e.Lock()
+	defer e.Unlock()
+
+	if s.Credentials.Passkey.Identifier {
+		e.Identifier = strings.ToLower(fmt.Sprintf("%s", value))
+	}
+
+	return nil
+}
+
+func (e *SchemaExtension) Finish() error { return nil }
+
+// PasskeyIdentifierFromIdentity returns the passkey identifier from the
+// identity. It is usually the email address and used to name the passkey in the
+// browser.
+func (s *Strategy) PasskeyIdentifierFromIdentity(ctx context.Context, id *identity.Identity) string {
+	e := new(SchemaExtension)
+	_ = s.d.IdentityValidator().ValidateWithRunner(ctx, id, e)
```

Comment:
Why do we ignore the error here?

Reply:
Proper validation already happens later. We just want the identifier node, and if it is empty, handle it properly on the call site.